### PR TITLE
CPU (Linux): use marketing name on Asahi Linux

### PR DESCRIPTION
**Pull request by ydalton**
I saw your reddit post on the Asahi Linux subreddit and wanted to help out. I admit it's not the best code but I don't think there is another way. Since you have an M1 Pro, can you check if it reports the correct CPU model?

### CODE REVIEW

File: src/detection/cpu/cpu_linux.c [hunk 1]
```diff
@@ -197,6 +197,86 @@ const char* ffDetectCPUImpl(const FFCPUOptions* options, FFCPUResult* cpu)
...
+                        cpu_model_name = "M3";
+                    }
+                    else if (strcmp(cpu_compat, "t6030") == 0)
+                    {
+                        cpu_model_name = "M3 Pro";
+                    }
+                    else if (strcmp(cpu_compat, "t6031") == 0
+                                    || strcmp(cpu_compat, "t3034") == 0)
+                    {
+                        cpu_model_name = "M3 Pro";
```
Comment (CarterLi) [inline src/detection/cpu/cpu_linux.c:267]: M3 Max?

File: src/detection/cpu/cpu_linux.c [hunk 2]
```diff
@@ -197,6 +197,86 @@ const char* ffDetectCPUImpl(const FFCPUOptions* options, FFCPUResult* cpu)
...
+            // In Asahi Linux, reading /proc/device-tree/compatible gives
+            // information on the device model. It consists of 3 NUL terminated
+            // strings, the second of which gives the actual SoC model. But it
+            // is not the marketing name, i.e. for M2 there is "apple,t8112" in
+            // the compatible string.
+            //
+            // A full list of the SoC model names can be found here:
+            // https://github.com/AsahiLinux/docs/wiki/Codenames
+            #define DT_COMPAT_PATH "/proc/device-tree/compatible"
+            if (strcmp(cpu->vendor.chars, "Apple") == 0)
```
Comment (CarterLi) [inline src/detection/cpu/cpu_linux.c:209]: ffStrbufEqualS(&cpu->vendor, "Apple")

Comment (CarterLi) [reply src/detection/cpu/cpu_linux.c:209]: Where do you get the vendor name?

File: src/detection/cpu/cpu_linux.c [hunk 3]
```diff
@@ -197,6 +197,86 @@ const char* ffDetectCPUImpl(const FFCPUOptions* options, FFCPUResult* cpu)
...
 
+            // In Asahi Linux, reading /proc/device-tree/compatible gives
+            // information on the device model. It consists of 3 NUL terminated
+            // strings, the second of which gives the actual SoC model. But it
+            // is not the marketing name, i.e. for M2 there is "apple,t8112" in
+            // the compatible string.
+            //
+            // A full list of the SoC model names can be found here:
+            // https://github.com/AsahiLinux/docs/wiki/Codenames
+            #define DT_COMPAT_PATH "/proc/device-tree/compatible"
```
Comment (CarterLi) [inline src/detection/cpu/cpu_linux.c:208]: Maybe we should guard this with `#ifdef __aarch64__`

File: src/detection/cpu/cpu_linux.c (hunk 1 above)
Comment (ydalton) [reply src/detection/cpu/cpu_linux.c:267]: Yep, that's an oversight on my part. It should be "M3 Max".

File: src/detection/cpu/cpu_linux.c (hunk 3 above)
Comment (ydalton) [reply src/detection/cpu/cpu_linux.c:208]: Sure.

File: src/detection/cpu/cpu_linux.c (hunk 2 above)
Comment (ydalton) [reply src/detection/cpu/cpu_linux.c:209]: The section above that runs `lscpu` and parses the  "Vendor ID" section correctly sets the vendor to "Apple". On my machine it looks like this:
```
Vendor ID:              Apple
  Model name:           Blizzard-M2
    Model:              0
```
I'll update to use the correct function.

**Pull request merged**